tools/serverpod_cli/lib/src/generator/dart/library_generators/library_generator.dart

```diff
@@ -1614,3 +1616,33 @@ extension on DatabaseDefinition {
     ]).code;
   }
 }
+
+/// Sorts child classes before their parents, such that serialization order is stable.
+int _byChildClassesBeforeParents(
+  SerializableModelDefinition a,
+  SerializableModelDefinition b,
+) {
+  if (a is! ModelClassDefinition || b is! ModelClassDefinition) {
+    return 0;
```

Comment:
🤔 .. seems odd

Reply:
Yes, I concluded that as well. Perhaps we could make capture that in the signature, or throw an exception here?

Anyway, not crucial.